### PR TITLE
logictest: deflake zone_config_system_tenant

### LINKED ISSUE #109117: pkg/sql/logictest/tests/local/local_test: TestLogic_zone_config_system_tenant failed
pkg/sql/logictest/tests/local/local_test.TestLogic_zone_config_system_tenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/11410867?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/11410867?buildTab=artifacts#/) on master @ [de0f60255e3f82da1b37504b294ae91f9268942d](https://github.com/cockroachdb/cockroach/commits/de0f60255e3f82da1b37504b294ae91f9268942d):

```
abandoned_vm:                       0
extent_avail:                       2
                           n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
large                              32       0               0       0               0       0              14       0               0         0               0           0
extent_avail                       40       0               0       0               0       0              18       0               0         0               0           0
extents_dirty                      38       0               0       0               0       0              18       0               0         0               0           0
extents_muzzy                      32       0               0       0               0       0              14       0               0         0               0           0
extents_retained                   40       0               0       0               0       0              18       0               0         0               0           0
decay_dirty                        36       0               0       0               0       0              17       0               0         0               0           0
decay_muzzy                        32       0               0       0               0       0              14       0               0         0               0           0
base                               69       0               0       0               0       0              15       0               0         0               0           0
tcache_list                        32       0               0       0               0       0              14       0               0         0               0           0
hpa_shard                           0       0               0       0               0       0               0       0               0         0               0           0
hpa_shard_grow                      0       0               0       0               0       0               0       0               0         0               0           0
hpa_sec                             0       0               0       0               0       0               0       0               0         0               0           0
bins:           size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests   (#/sec)  nshards      curregs     curslabs  nonfull_slabs regs pgs   util       nfills (#/sec)     nflushes (#/sec)       nslabs     nreslabs (#/sec)      n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
                     ---
large:          size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests (#/sec)  curlextents
                     ---
             8388608  72            0            3       0            3       0            3       0            0
                     ---
extents:        size ind       ndirty        dirty       nmuzzy        muzzy    nretained     retained       ntotal        total
                     ---
            10485760  40            0            0            0            0            1     10485760            1     10485760
            12582912  41            0            0            0            0            1     12582912            1     12582912
                     ---
Bytes in small extent cache: 0
HPA shard stats:
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 

                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---
[07:41:06] --- total progress: 31 statements
--- total: 31 tests, 2 failures
    --- FAIL: TestLogic_zone_config_system_tenant/dropped_relation_gc_ttl (3.24s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_zone_config_system_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-30789

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/109117
Release note: None

### CODE REVIEW

**Review by arulajmani — APPROVED** [review]
:LGTM: modulo comment.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109376#-:-NcZQTWF-RE74b5U0xvH:b4linka)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant` line 155 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/109376#-NcZPuGACvNlVyXoBf8R:-NcZPuGACvNlVyXoBf8S:b-5bpuhw) ([raw file](https://github.com/cockroachdb/cockroach/blob/4fa978c87ff5e7c61f44ac990697ef0977b85330/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant#L155)):*
> ```
> # have a gc_ttl of 14400.
> query T retry
> SELECT crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)->'gcPolicy'->>'ttlSeconds'
> ```

Should we modify these queries to include the start key as well? (there's a `crdb_internal.pretty_key` builtin that we could use to make them readable).

As is, like we found, this output is really hard to parse when the test fails because all it's doing is selecting the GC TTL. We could even consider doing that in other places we're querying from `system.span_configurations` in this file.

**Review by rafiss — COMMENTED** [review]
tftr!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109376#-:-NcZWqlW0Pm372lSBoU3:bjpalj3)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani)

___
*[`pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant` line 155 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/109376#-NcZPuGACvNlVyXoBf8R:-NcZWpNXElMv3GmvV0Rc:b-93ra9t) ([raw file](https://github.com/cockroachdb/cockroach/blob/4fa978c87ff5e7c61f44ac990697ef0977b85330/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant#L155)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Should we modify these queries to include the start key as well? (there's a `crdb_internal.pretty_key` builtin that we could use to make them readable).

As is, like we found, this output is really hard to parse when the test fails because all it's doing is selecting the GC TTL. We could even consider doing that in other places we're querying from `system.span_configurations` in this file.
</blockquote></details>

great idea. added

**Comment by yuzefovich** [comment]
blathers backport 23.1
